### DISCUSSION TITLE
Block Not Found error on waitForTransaction

### DISCUSSION THREAD

**Discussion by albydarned** [WalletConnect/walletconnect-monorepo#3736] [Developer Support] [answered]
Recently I switched from an ethers centered injected wallet connection to walletconnect in my dApp. For background, my dapp uses the polygon mumbai testnet. I am a Vue.js app using Web3Modal v3 as follows:

```// WalletConnect
    import { createWeb3Modal, defaultWagmiConfig } from '@web3modal/wagmi/vue'
    import { polygonMumbai } from '@wagmi/core/chains'
    import { watchAccount } from '@wagmi/core'
    // Get projectId
    const projectId = import.meta.env.VITE_WALLETCONNECT_ID;
    // Create wagmiConfig
    const chains = [polygonMumbai]
    const wagmiConfig = defaultWagmiConfig({ chains, projectId, appName: 'dApp' })
    // Create modal
    const web3Modal = createWeb3Modal({ 
      wagmiConfig, projectId, chains,
      themeMode: 'light',
      themeVariables: {
        '--w3m-accent': '#0AA7FC'
      }
    });
```

Everything works great 90% of the time, but occasionally and randomly I get a "Block Not Found" error when using waitForTransaction. It appears to be the exact same issue as this inside of wagmi:
https://github.com/wagmi-dev/wagmi/issues/2530 

In my WalletConnect dashboard under project metrics I can see my RPC coming through. I'm guessing whats happening is the provider being passed to wagmi has aggressive load balancing and while it can fetch the block for a txn it can't actually await the txn. I never had this issue when using an injected Infura provided wallet. It does seem upping the confirmations on waitForTransaction has helped, but not alleviated the issue. Previously I was just waiting for 1 confirmation and that was fine.

Has anyone else run into this? Could I perhaps pass my own Infura based provider in? There aren't a ton of docs yet for walletconnect v3. I saw some reference to this in google searches, but the actual pages 404.

**Comment by glitch-txs** [comment, marked as answer]
Hi, you can use Custom configuration without the `defaultWagmiConfig` function and add other providers 
- Check for custom config: https://docs.walletconnect.com/web3modal/react/about
- Add more providers: https://wagmi.sh/react/providers/configuring-chains

We also recommend adding the walletConnectProvider

```ts
import { walletConnectProvider } from '@web3modal/wagmi'
```

**Reply by glitch-txs** [reply]
I've also created this feature request in Web3Modal repo, would love your feedback if you think this would be good DX for the defaultWagmiConfig
https://github.com/WalletConnect/web3modal/issues/1373

**Reply by albydarned** [reply]
Thank you very much

**Comment by albydarned** [comment]
For those needing an example of how this can be done, this works in my Vue app

```
    // WalletConnect
    import { createWeb3Modal } from '@web3modal/wagmi/vue'
    import { walletConnectProvider } from '@web3modal/wagmi'
    import { configureChains, createConfig, watchAccount } from '@wagmi/core'
    import { polygonMumbai } from '@wagmi/core/chains'
    import { publicProvider } from '@wagmi/core/providers/public'
    import { infuraProvider } from '@wagmi/core/providers/infura'
    import { CoinbaseWalletConnector } from '@wagmi/core/connectors/coinbaseWallet'
    import { InjectedConnector } from '@wagmi/core/connectors/injected'
    import { WalletConnectConnector } from '@wagmi/core/connectors/walletConnect'

    // Define WalletConnect metadata
    const metadata = {
        name: 'Web3Modal',
        description: 'Web3Modal Example',
        url: 'https://web3modal.com',
        icons: ['https://avatars.githubusercontent.com/u/37784886']
      }
    // Import env keys
    const projectId = import.meta.env.VITE_WALLETCONNECT_ID;
    const infuraKey = import.meta.env.VITE_INFURA_KEY;
    // Get chains and publicClient
    const { chains, publicClient } = configureChains(
      [polygonMumbai],
      [
        infuraProvider({ apiKey: infuraKey }),
        walletConnectProvider({ projectId: projectId }),
        publicProvider(),
      ]
    );
    // Create wagmi config
    const wagmiConfig = createConfig({
      autoConnect: true,
      connectors: [
        new WalletConnectConnector({ chains, options: { projectId, showQrModal: false, metadata } }),
        new InjectedConnector({ chains, options: { shimDisconnect: true } }),
        new CoinbaseWalletConnector({ chains, options: { appName: metadata.name } })
      ],
      publicClient
    });
    // Create modal
    const web3Modal = createWeb3Modal({ 
      wagmiConfig, projectId, chains,
      themeMode: 'light',
      themeVariables: {
        '--w3m-accent': '#0AA7FC'
      }
    });
```
